### ISSUE TITLE
Use a mock of pyminiaudio in functional tests

### ISSUE THREAD

**Issue by postlund** [postlund/pyatv#1437]
### Short feature/function description

There's currently a **big** problem with stability in tests which seems to be related to using pyminiaudio. I'm not sure what the reason behind it is, but I suspect it has to do with cffi ("Illegal instruction"). I also suspect that system load is a factor.

### What needs to be done?

I intent to create a mock of pyminiaudio and use that in the tests and possibly write *one* integration test that verifies using the library. Hopefully that make things more stable and also faster.

### Is this a breaking change?

No

### Anything else worth knowing?

No

**Comment by postlund** [comment]
Tests are so unstable right now so this issue has highest priority. Will start tomorrow.

**Comment by postlund** [comment]
I have tried replacing tests with a mock, leaving only two integration tests for the library and one functional test that I couldn't use a mock in. Still no dice, I get crashes. Also tried to exclude all tests using miniaudio and run them sequentially (one process) but still get crashes. Only with GitHub actions though... Not really sure what to do from here.

**Comment by postlund** [comment]
It's easy to see where it happens now:

```
Fatal Python error: Illegal instruction

Current thread 0x00007f1d9628a700 (most recent call first):
  File "/home/runner/.virtualenvs/.venv/lib/python3.10/site-packages/miniaudio.py", line 931 in decode_file
  File "/opt/hostedtoolcache/Python/3.10.0/x64/lib/python3.10/concurrent/futures/thread.py", line 52 in run
  File "/opt/hostedtoolcache/Python/3.10.0/x64/lib/python3.10/concurrent/futures/thread.py", line 77 in _worker
...
```

**Comment by postlund** [comment]
Ok, so, I took a different route here... I remembered that the same issue has been seen for people running on Raspberry pi. General solution was to re-install pyminaudio and build from source. I am now doing that in the tests workflow. A step named *miniaudio* has been added to to chickn and tagged with miniaudio as well (so not run by default). I've run this twice and got successful builds both times (without any cached jobs). So promising so far, but time will tell if it works.

Because of this I have also decided to not move forward with any other changes. Keep it small and as simple as possible.
